features/networking/ovn.feature

```diff
@@ -0,0 +1,147 @@
+Feature: OVN related networking scenarios
+
+  # @author anusaxen@redhat.com
+  # @case_id OCP-29954
+  @admin
+  @destructive
+  Scenario: Creating a resource in Kube API should be synced to OVN NB db correctly even post NB db crash too
+    Given the env is using "OVNKubernetes" networkType
+    Given I have a project
+    And I obtain test data file "networking/list_for_pods.json"
+    When I run the :create client command with:
+      | f | list_for_pods.json |
+    Then the step should succeed
+    Given 2 pods become ready with labels:
+      | name=test-pods |
+    And evaluation of `pod(0).name` is stored in the :pod1_name clipboard
+    And evaluation of `pod(1).name` is stored in the :pod2_name clipboard
+    #Checking whether Kube API data is synced on OVN NB db which in this case are couple of pods
+    Given I store the ovnkube-master "north" leader pod in the clipboard
+    And evaluation of `pod.ip_url` is stored in the :ovn_nb_leader_ip clipboard
+    And evaluation of `pod.node_name` is stored in the :ovn_nb_leader_node clipboard
+    And admin executes on the pod:
+      | bash | -c | ovn-nbctl list logical_switch_port |
+    Then the step should succeed
+    And the output should contain:
+      | <%= cb.pod1_name %> |
+      | <%= cb.pod2_name %> |
+    #Simulating a NB db crash
+    Given I use the "<%= cb.ovn_nb_leader_node %>" node
+    And I run commands on the host:
+      | pkill -f OVN_Northbound |
+    And admin waits for all pods in the "openshift-ovn-kubernetes" project to become ready up to 120 seconds
+    #Making sure the pod entries are synced again when NB db is re-created
+    Given I store the ovnkube-master "north" leader pod in the clipboard
+    And evaluation of `pod.ip_url` is stored in the :new_ovn_nb_leader_ip clipboard
+    Given admin executes on the pod:
+      | bash | -c | ovn-nbctl list logical_switch_port |
+    Then the step should succeed
+    And the output should contain:
+      | <%= cb.pod1_name %> |
+      | <%= cb.pod2_name %> |
+
+
+  # @author anusaxen@redhat.com
+  # @case_id OCP-30055
+  @admin
+  @destructive
+  Scenario: OVN DB should be updated correctly if a resource only exist in Kube API but not in OVN NB db
+    Given the env is using "OVNKubernetes" networkType
+    # Now scale down CNO pod to 0
+    Given admin uses the "openshift-network-operator" project
+    And I run the :scale admin command with:
```

Comment:
do we need to scale the network operator to 0 ? is there any difference if only delete the `ovnkube-master ` pods and make it recreated ?

Reply:
ok, thanks for your explanation.  I thought they should be have same behaviour (with deleting cno) and (without deleting cno).
anyways. if you already confirmed this with Dev. that's ok for me.

Reply:
scaling to zero makes sense.

If we do need to do a more fine-grained pause, we could find the exact processes and `pkill -STOP`  which freezes the process in the kernel scheduler without killing it, then `pkill -CONT` to resume.

For the northd leader election lock inactivity timer BZ-1834473, I tested with `pkill -STOP` and `pkill -CONT` to check the inactivity timeout, so at some point we will have a test for that.

Reply:
@rbbratta yea, thats a good suggestion for processes but as we know CNO runs as container so we cant really leverage that here. But this can be used for NB db which runs as a process. But again IMO pkill would make sense as we are simulating a intentional NB db crash here which should be quickly resilient.